kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/KafkaUtils.java

```diff
@@ -45,56 +45,7 @@ public class KafkaUtils {
     private static final String MESSAGE_VAR = "%MESSAGE%";
     private static final String KAFKA_VERSION_VAR = "%KAFKA_VERSION%";
 
-    /**
-     * Consume message string.
-     *
-     * @param deployNamespace the deploy namespace
-     * @param topicName the topic name
-     * @param bootstrap the bootstrap
-     * @param timeout the timeout
-     * @return the log of the pod
-     */
-    public static String consumeMessage(String deployNamespace, String topicName, String bootstrap, Duration timeout) {
-        LOGGER.debug("Consuming messages from '{}' topic", topicName);
-
-        String kafkaConsumerName = "java-kafka-consumer";
-        Pod pod = kubeClient().getClient().run().inNamespace(deployNamespace).withNewRunConfig()
-                .withImage(Constants.STRIMZI_KAFKA_IMAGE)
-                .withName(kafkaConsumerName)
-                .withRestartPolicy("Never")
-                .withCommand("/bin/sh")
-                .withArgs("-c",
-                        "bin/kafka-console-consumer.sh --bootstrap-server " + bootstrap + " --topic " + topicName + " --from-beginning --timeout-ms "
-                                + timeout.toMillis())
-                .done();
-        long deadline = System.currentTimeMillis() + timeout.toMillis() * 2L;
-        long timeLeft = deadline;
-        while (timeLeft > 0) {
-            try {
-                Thread.sleep(1000);
-                timeLeft = deadline - System.currentTimeMillis();
-            }
-            catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                LOGGER.trace(e.getMessage());
-            }
-        }
-        String log = kubeClient().logsInSpecificNamespace(deployNamespace, kafkaConsumerName);
-        kubeClient().getClient().pods().inNamespace(deployNamespace).withName(kafkaConsumerName).delete();
-        return log;
-    }
-
-    /**
-     * Consume message with test clients.
-     *
-     * @param deployNamespace the deploy namespace
-     * @param topicName the topic name
-     * @param bootstrap the bootstrap
-     * @param numOfMessages the num of messages
-     * @param timeout the timeout
-     * @return the log of the pod
-     */
-    public static String consumeMessageWithTestClients(String deployNamespace, String topicName, String bootstrap, int numOfMessages, Duration timeout) {
+    private static String consumeMessages(String deployNamespace, String topicName, String bootstrap, int numOfMessages, String messageToSeek, Duration timeout) {
```

Comment:
'seek' in Kafka has meaning already, so your use of it in this way misleads the right.

Reply:
So this is a good usecase for a java.util.function.Predicate.  I'd suggest messageToSeek becomes a messagePredicate

Reply:
renamed